app/src/main/java/com/github/sdp/mediato/ui/MyProfileFragment.java

```diff
@@ -68,54 +60,59 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
 
         // Initialize components
         photoPicker = setupPhotoPicker();
-        collectionlistAdapter = setupCollections(collectionListRecyclerView);
         setupAddCollectionsButton(addCollectionButton);
 
-        // Observe the view model's live data to update UI components
-        observeCollections(collectionlistAdapter);
+        // Set up collections
+        fetchCollectionsFromDatabaseWithRetry(0).thenAccept(collections -> {
+            collectionlistAdapter = setupCollections(collectionListRecyclerView, collections);
+            // Observe the view model's live data to update UI components
+            observeCollections(collectionlistAdapter);
+            // Add a review if there is one
+            addReview();
+        });
 
         // Add on click listener to sign out button
         Button signOutButton = view.findViewById(R.id.signout_button);
         signOutButton.setOnClickListener(v -> ((MainActivity) getActivity()).signOutUser());
 
-        // Add a review if there is one
-        addReview();
-
         return view;
     }
 
     @Override
-    public CollectionListAdapter setupCollections(RecyclerView recyclerView) {
-        // Check if a collection is already in the viewModel, if not create the default one
-        List<Collection> collections = viewModel.getCollections();
-        if (collections == null) {
-            collections = createDefaultCollection();
-        }
+    public CollectionListAdapter setupCollections(RecyclerView recyclerView,  List<Collection> collections) {
+        viewModel.setCollections(collections);
 
         // Define what happens when the add button inside a collection is clicked
```

Comment:
I think we can get rid of the comments next to the button click listeners. The rest of the code is already full of comments, it would be nice to not overload it.

Reply:
done